corehq/apps/userreports/sql/adapter.py

```diff
@@ -452,12 +452,12 @@ def get_indicator_table(indicator_config, metadata, override_table_name=None):
             logger.error(f"Invalid index specified on {table_name} ({index.column_ids})")
     constraints = [PrimaryKeyConstraint(*indicator_config.pk_columns)]
     columns_and_indices = sql_columns + extra_indices + constraints
-    # todo: needed to add extend_existing=True to support multiple calls to this function for the same table.
-    # is that valid?
+    current_table = metadata.tables[table_name]
+    if current_table:
+        metadata.remove(current_table)
```

Comment:
For those of us not super familiar with sqlalchemy tables and metadata—what do these lines do?

Reply:
`MetaData` is basically a python object description of a database. It is primarily a collection of tables and their related objects. That DB may or may not exist in an actual database. We use it in the pillow to construct a model of what we believe the UCR DB structure should be, and then compare that to a model created by actually checking the tables in postgres, and make any necessary changes to the tables in postgres.

When we create the table with `extend_existing`, it checks for a table with that name in the existing metadata, and adds any newly defined columns to the existing definition, rather than replacing the existing definition with the new one. This change checks if the relevant table already exists in the `MetaData`, and if so, deletes it, so that when we create the table object it cleanly replaces the old one.